### ISSUE TITLE
Rails Cop Idea: `Rails.root.join`

### ISSUE THREAD

**Issue by mockdeep** [bbatsov/rubocop#3779]
Would love a cop that specifies to always use syntax:

```ruby
Rails.root.join('app', 'models', 'goober') # good
```

over:

```ruby
# bad
Rails.root.join('app/models/goober')
"#{Rails.root}/app/models/goober"
File.join(Rails.root, 'app/models/goober')
# etc...
```

We've got a bunch of formats littered throughout our codebase.

**Comment by bbatsov** [comment]
Seems reasonable to me.

**Comment by rrosenblum** [comment]
I like the idea of this cop. There is some room to cover functionality from `File.join` as well. From your example, the variable length args is preferred over a single argument. This would also work with `File.join`. We would probably need to be able to support both styles (variable length args and single arg).

```ruby
# single arg
Rails.root.join('app/models/goober')
File.join('app/models/goober')

# var arg
Rails.root.join('app', 'models', 'goober')
File.join('app', 'models', 'goober')
```

One thing to be careful with is how `Rails.root.join` handles an argument that begins with '/'. 
```ruby
[9] pry(main)> File.join('app', 'models')                                                                                                                                         
=> "app/models"
[10] pry(main)> File.join('app', '/models')
=> "app/models"

[7] pry(main)> Rails.root.join('app', 'models')
=> #<Pathname:/home/username/work/project/app/models>
[8] pry(main)> Rails.root.join('app', '/models')
=> #<Pathname:/models>
```

**Comment by roberts1000** [comment]
Out of curiosity, why is the cop recommended?  I couldn't find anything in https://rubocop.readthedocs.io/en/latest/cops_rails/#railsfilepath or https://github.com/bbatsov/rails-style-guide that talks about the rationale behind it.  

A statement like

```ruby
require "#{Rails.root}/app/models/foo"
```

seems easier to read, and faster, than

```ruby
require Rails.root.join('app', 'models', 'foo')
```

**Comment by mikegee** [comment]
@roberts1000 `require Rails.root.join('app', 'models', 'foo')` works on Windows and other operating systems that don't use `"/"` as the path separator.

It is unfortunate that none of the PR, Issue, documentation, code comments, or offense message justify this cop.

**Comment by roberts1000** [comment]
@mikegee Thanks for the explanation!

**Comment by bbatsov** [comment]
@mikegee I see a PR opportunity here! ;-) (for both RuboCop and the Rails style guide)

**Comment by KurtRMueller** [comment]
But now rubocop is complaining that:
`Rails/FilePath: Please use Rails.root.join('path/to') instead`

Why the change?

**Comment by sunny** [comment]
@KurtRMueller It is a common misconception that `Rails.root.join('app/models/foo')` does not work on Windows and other operating systems that don't use "/" as the path separator. Check out for https://github.com/rubocop-hq/rubocop/issues/5823 for the issue that led to the change.
